File: 507.perfect-number.java
/*
 * @lc app=leetcode id=507 lang=java
 *
 * [507] Perfect Number
 */

// @lc code=start
class Solution {
    public boolean checkPerfectNumber(int num) {
        if (num == 1) {
            return false;
        }
        long sum = 1;
        for (int i = 2; i<= Math.sqrt(num); i++) {
            if (num % i == 0) {
                sum += i;
                if (i * i != num) {
                sum += num/i;
                }
            }
        }
        if (num == sum) {
        return true;
        }
        return false;
    }
}
// @lc code=end

File: 507.perfect-number.py
#
# @lc app=leetcode id=507 lang=python
#
# [507] Perfect Number
#

# @lc code=start
class Solution(object):
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num == 1:
            return False
        factors = 1
        og = num
        for i in range(2,int(num**0.5)+ 1):
            if num%i == 0:
                factors += i
                if i*i != num:
                    factors += num//i
        if factors == og:
            return True
        else:
            return False
        
        
# @lc code=end

